### DISCUSSION TITLE
installation

### DISCUSSION THREAD

**Discussion by attila2005** [copymark/TM-Trainer#11] [Q&A] [answered]
this is probably a stupid question, but i dont really know how to install this thing xD could someone help me

**Comment by copymark** [comment]
Get Visual Studio Comunity 2019 and load the sln file from the root directory. Build the Solution in RELEASE mode.

**Comment by Ruitjes** [comment]
@attila2005  Dont even bother unless you really want to update this, because this is not working anymore and is outdated.

**Reply by copymark** [reply]
Tm nations/United should still work afaik

**Comment by attila2005** [comment]
I dont know how to do any of this, i downloaded visual studio and im too dumb to get it to work xd is there an already a working TMUF Trainer that i can download without having to do this stuff? i just need the TMUF Trainer, nothing else

**Comment by attila2005** [comment]
hello? anyone

**Comment by copymark** [comment, marked as answer]
[TM Ultra Trainer - copymark - 07.04.2016.zip](https://github.com/copymark/TM-Trainer/files/5721660/TM.Ultra.Trainer.-.copymark.-.07.04.2016.zip)
attila2005
BCAU-FWGS-QFG5-WUW7

**Comment by attila2005** [comment]
Thank you very much 👍

**Comment by man10manusa** [comment]
can work Trainer TrackMania² Canyon ?? plzzzzzzzzzzzz
